### ISSUE TITLE
CAPTCHA was not successful. Please try again. v.2.1.17

### ISSUE THREAD

**Issue by dess98** [geoffhumphrey/brewcompetitiononlineentry#1048]
Version:

Installation URL:

Is your installation hosted on brewcompetition.com or brewcomp.com?

Description of Issue:

For enhancements, please prepend "Enhancement - " to the title. For issues/bugs, please prepend the BCOE&M version number the title.

**Comment by geoffhumphrey** [comment]
A bit more information would be helpful. What URL? Since CAPTCHA is a Google plugin, the circumstances will very much help to track down the cause.

**Comment by dess98** [comment]
Sorry, I did explain more but must have not clicked save or something because none of that is here in my post. I shall try again . . . Our url is njhopz.com/dasboot. We are hosting through our site. We are running the latest version - 2.1.17. 

We have not been able to register anyone. On at least 3 different computers and my phone, every time we have tried to register a judge or steward the captcha fails. 

When we first installed we would see the word "CAPTCHA" above the register button. The nessage "CAPTCHA was not successful. Please try again." would appear on a new page. So we tried creating our own google captcha, entering the keys in preferences as directed, and trying again. While we now do see a captcha checkbox, when we check it, get a checkmark, and click register, we get the same message on a new page "CAPTCHA was not successful. Please try again."

I really don't want to bypass recaptcha but I just don't know what else I can do.
I'm hoping there is some kind of work-around, registration for entrants opens Monday.

Thank you for your help.

**Comment by Jwwilson3** [comment]
Any news on this as we are experiencing the same.  When we first uploaded the software we couldn't even see the captcha box.  We installed new keys, went trhough registration, saw the captcha checkbox, went through the test and then received an error "Captcha was not successful. Please try again."  That is as far as we got.  I started over, deleted the keys and installed new keys a dozen times to no avail.  I am opening registration at midnight and will have to turn off the captcha unless there is a fix I have not tried?  How detrimental is not having Captcha?

**Comment by geoffhumphrey** [comment]
In all cases, it's essential the domain be added to the reCAPTCHA settings page (via Google) that's associated with your site and secret keys.

![2019-04-01_11-18-22](https://user-images.githubusercontent.com/5216502/55346599-1bbdba80-5470-11e9-8fcc-05bbab0f213b.png)

The biggest detriment with not having CAPTCHA is the risk of bots registering an account and sending spam emails via the contact forms.

**Comment by dess98** [comment]
Yes, we followed the directions for the google recaptcha and registered our domain name as stated. We even tried using that recaptcha elsewhere on our website and it worked fine. So we obviously did the recaptcha correctly, and the problem is with some kind of conflict with the bcoem.

**Comment by geoffhumphrey** [comment]
Unfortunately, I'm not able to recreate the issue on any of my test environments. If you have the ability, perhaps there's something in the server logs that may shed light on why. My thought is that your server isn't allowing the recaptcha relay with Google for verification.

Also, if you wouldn't mind posting the URL to your install, I'll take a look using dev tools.

**Comment by dess98** [comment]
Sure, the url is https://njhopz.com/dasboot/.

I have looked in the error logs, is that the same thing? Would it be
helpful for me to copy, say, the errors from Sunday when we were trying
over and over, into an email for you to see? Is there some other way I can
give you access?

Thank you for your help,
Maia

**Comment by geoffhumphrey** [comment]
@dess98 - yes, that would be good. Go ahead and send the log to prost [at] brewcompetition.com and I see if anything stands out. 

Otherwise, for me to test via dev tools, you'll need to activate reCAPTCHA via the site preferences. Understandable if you don't wish to do that while you're accepting accounts and entries, however.

**Comment by geoffhumphrey** [comment]
@dess98 - apologies for the delay, but I was able to take a look at the server logs you sent over and found that a php setting is disabled that is needed for communication with Google's API. 

Have your web server administrator make sure to enable allow_url_fopen in your server php.ini file (or settings via cpanel). PHP's default value is 1 (allow), but it appears your server is configured to 0 (disallow). 

For more info:
https://www.php.net/manual/en/filesystem.configuration.php#ini.allow-url-fopen

**Comment by dess98** [comment]
Hi Geoff,

I finally had a chance to get into the cpanel and take a look.  According
to what is shown in the MultiPHP INI Editor in cpanel, allow_url_fopen is
enabled.  So I'm not sure that will make any difference.  Any other
possibilities?

Thanks, Maia

**Comment by dess98** [comment]
I should mention that I was asking a friend to help who was looking at the
error logs and suggesting some changes I should make.  I may have enabled
the allow_url_fopen at some point on the 31st.  (There were several things,
I definitely used the MultiPHP INI Editor to do something.)  I tried
repeatedly after making all changes but the recaptcha continued to fail and
not allow me to register.

Thanks, Maia

**Comment by grmelacz** [comment]
As we had previously problems with reCAPTCHA, we had to disable it. Since there was a lot of spam incoming to the contact form in the past week, I've enabled it. Created a v2 checkbox captcha at Google, entered the keys to BCOE&M and everything worked on the first try without any problems. I do not have chance to test the registration though as we've just finished our competition yesterday (only tested on the contact form).
